### PR TITLE
[DataFactory]Support New Features In ADF

### PR DESCRIPTION
**Pull request by Jingshu918**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/DataFactory/DataFactory.Management.Sdk/Generated/Models/MariaDBLinkedServiceTypeProperties.cs [hunk 1]
```diff
@@ -138,6 +172,8 @@ public virtual void Validate()
 
 
 
+
```
Comment (msJinLei) [inline src/DataFactory/DataFactory.Management.Sdk/Generated/Models/MariaDBLinkedServiceTypeProperties.cs:175]:
```suggestion
```

File: src/DataFactory/DataFactory.Management.Sdk/Generated/Models/MariaDBLinkedServiceTypeProperties.cs [hunk 2]
```diff
@@ -138,6 +172,8 @@ public virtual void Validate()
...
+
```
Comment (msJinLei) [inline src/DataFactory/DataFactory.Management.Sdk/Generated/Models/MariaDBLinkedServiceTypeProperties.cs:176]:
```suggestion
```

File: src/DataFactory/DataFactory/ChangeLog.md [hunk 3]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Added support for Iceberg format as a sink.
```
Comment (msJinLei) [inline src/DataFactory/DataFactory/ChangeLog.md:21]: Do you have the help document update?

Comment (Jingshu918) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: No, ADF only update sdk for powershell, nothing change in cmd

Comment (msJinLei) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: > No, ADF only update sdk for powershell, nothing change in cmd

@Jingshu918 Please confirm you module is not going to release a new version after the merge is merged

Comment (Jingshu918) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: I don't know what you mean "module is not going to release a new version", After the PR merged Powershell should release a new version for DataFactory, for such change we have released many years.

https://github.com/Azure/azure-powershell/pull/25825
https://www.powershellgallery.com/packages/Az.DataFactory/1.18.7

Comment (msJinLei) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: @Jingshu918 Sorry, please ignore the previous comment. I may comment in the wrong PR.
The review comments for you PR is
Please refer to https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md to update your help doc of your module

Comment (Jingshu918) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: The link is for powershell autorest track 2, this is not we are using. And we are not updating our module, why need update the docs?

Comment (Jingshu918) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: We are not updating our modules, so we need not do that.

Comment (msJinLei) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: > We are not updating our modules, so we need not do that.

The help doc should be consistent with the source codes and currently the cmdlet help docs are already out of date.

Comment (Jingshu918) [reply src/DataFactory/DataFactory/ChangeLog.md:21]: I don't think it is out of data in DataFactory. DataFactory maybe a special Service from others.
For example, we only change the definitionFile which customers can know from other places and it did not need customer change the modules and docs.

Set-AzDataFactoryV2Dataset
   [-Name] <String>
   [-DefinitionFile] <String>
   [-ResourceGroupName] <String>
   [-DataFactoryName] <String>
   [-Force]
   [-DefaultProfile <IAzureContextContainer>]
   [-WhatIf]
   [-Confirm]
   [<CommonParameters>]
   
   https://learn.microsoft.com/en-us/powershell/module/az.datafactory/set-azdatafactoryv2dataset?view=azps-12.3.0
   
   https://learn.microsoft.com/en-us/azure/data-factory/

**Comment by Jingshu918** [comment]
@msJinLei  Sorry, I did not know what you mean "out of date", do we have similar change for that?

**Comment by msJinLei** [comment]
> @msJinLei Sorry, I did not know what you mean "out of date", do we have similar change for that?

@Jingshu918 To update help doc is part of the development process of Azure-PowerShell. We usually ask the partners to update the help docs of their modules in their PRs even the docs changes are not caused by their PR merely. Are you the module owner of data factory? If not, could you provide the owner of `Data Factory` (usually a Microsoft FTE) and I will ask them to update the help docs of `Data Factory`.

**Comment by Jingshu918** [comment]
@msJinLei  I did not find you in teams, Could we create a team chart to discuss the docs?
Please add Ruoxuan(ruowan@microsoft.com) and me(v-jinpan@microsoft.com)

**Review by msJinLei — APPROVED** [review]

**Pull request merged**